spec/features/user_management_spec.rb

```diff
@@ -0,0 +1,60 @@
+require_relative '../factories/user'
+
+feature 'User sign up' do
+  scenario 'I can sign up as a new user' do
+    user = build :user
+    expect { sign_up(user) }.to change(User, :count).by(1)
+    expect(page).to have_content('Welcome, foo@bar.com')
+    expect(User.first.email).to eq('foo@bar.com')
+  end
+
+  scenario 'requires a matching confirmation password' do
+    user = build :baduser
+    expect { sign_up(user) }.not_to change(User, :count)
+  end
+
+  scenario 'with a password that does not match' do
+    user = build :baduser
+    expect { sign_up(user) }.not_to change(User, :count)
+    expect(current_path).to eq('/users')
+    expect(page).to have_content 'Password does not match the confirmation'
+  end
+
+  scenario 'I cannot sign up with an existing email' do
+    user = build :user
+    user2 = build :user
+    sign_up(user)
+    expect { sign_up(user) }.to change(User, :count).by(0)
+    expect(page).to have_content('Email is already taken')
+  end
+
+  feature 'User sign in' do
+    let(:user) do
+      User.create(email: 'user@example.com',
+                  password: 'secret1234',
+                  password_confirmation: 'secret1234')
+    end
+
+    scenario 'with correct credentials' do
+      sign_in(email: user.email,   password: user.password)
+      expect(page).to have_content "Welcome, #{user.email}"
+    end
+  end
+
+  feature 'User signs out' do
+
+    let(:user) do
+      User.create(email: 'test@example.com',
+                  password: 'secret1234',
+                  password_confirmation: 'secret1234')
+  end
+
+  scenario 'while being signed in' do
+      sign_in(email: 'test@example.com', password: 'test')
+      click_button 'Sign out'
+      expect(page).to have_content('goodbye!')
+      expect(page).not_to have_content('Welcome, test@example.com')
+    end
```

Comment:
Lint/BlockAlignment: `end` at 57, 4 is not aligned with `scenario 'while being signed in' do` at 52, 2